src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -38,10 +38,52 @@ import TestSQLContext._
 private[avro] object TestUtils {
 
   /**
+   * This function checks that all records in a file match the original
+   * record.
+   */
+
+  def checkReloadMatchesSaved(testFile: String, avroDir: String) = {
+
+    def convertToString(elem: Any): String = {
+      elem match {
+        case null => "NULL" // HashSets can't have null in them, so we use a string instead
+        case arrayBuf: ArrayBuffer[Any] => arrayBuf.toArray.deep.mkString(" ")
+        case arrayByte: Array[Byte] => arrayByte.deep.mkString(" ")
+        case other => other.toString
+      }
+    }
+
+    val originalEntries = TestSQLContext.avroFile(testFile).collect()
+    val newEntries = TestSQLContext.avroFile(avroDir).collect()
+
+    assert(originalEntries.size == newEntries.size)
+
+    val origEntrySet = new Array[HashSet[Any]](originalEntries(0).size)
+    for (i <- 0 until originalEntries(0).size) { origEntrySet(i) = new HashSet[Any]() }
```

Comment:
By the way, `Array.fill(originalEntries(0).size)(new HashSet[Any]())` would be the canonical way to do this